src/instrumentsscene/qml/MuseScore/InstrumentsScene/InstrumentsDialog.qml

```diff
@@ -55,7 +62,7 @@ StyledDialogView {
 
             canSelectMultipleInstruments: root.canSelectMultipleInstruments
             currentInstrumentId: root.currentInstrumentId
-
+            invoker: root
```

Comment:
And here, you would listen for the signal, and handle it:
```qml
    onSubmitRequested: {
        root.submit()
    }
```

This way, the ChooseInstrumentsPage does not depend on specific characteristics of any of its parent items ("characteristics" as in having a `submit()` function). But it just tells the parent that something happened, and the parent can act as it sees fit.

You will notice this pattern of having `...Requested()` signals very often in our QML code. It is good to use similar patterns whenever you need to communicate something upwards in the view hierarchy (from child component to parent component).